### PR TITLE
Fixes to refresh tokens.

### PR DESCRIPTION
**Pull request by mshima**
---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by mshima** [comment]
@mraible I couldn't get to work reactive to redirect.

File: generators/server/templates/src/main/java/package/config/OAuth2Configuration.java.ejs [hunk 1]
```diff
@@ -16,10 +18,21 @@ public class OAuth2Configuration {
...
             authorizedClientRepository
         );
+
+        authorizedClientManager.setAuthorizedClientProvider(
+            <%= reactivePrefix %>OAuth2AuthorizedClientProviderBuilder
+                .builder()
+                .authorizationCode()
+                .refreshToken(builder -> builder.clockSkew(Duration.ofMinutes(1)))
+                .clientCredentials()
+                .password()
```
Comment (mraible) [inline generators/server/templates/src/main/java/package/config/OAuth2Configuration.java.ejs:32]: This looks strange. Why do we need a password here and where does it come from?

@bdemers Can you have a look at this? We're trying to get refresh tokens working in JHipster.

Comment (mshima) [reply generators/server/templates/src/main/java/package/config/OAuth2Configuration.java.ejs:32]: It's the default for the DefaultOAuth2AuthorizedClientManager
https://github.com/spring-projects/spring-security/blob/006b9b960797d279b31cf8c8d16f1549c5632b2c/oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/web/DefaultOAuth2AuthorizedClientManager.java#L91-L96

I've added to let the user easily change clockSkew, but can be removed.

File: generators/server/templates/src/main/java/package/web/filter/OAuth2ReactiveRefreshTokensWebFilter.java.ejs [hunk 2]
```diff
@@ -46,6 +46,7 @@ public class OAuth2ReactiveRefreshTokensWebFilter implements WebFilter {
             .filter(principal -> principal instanceof OAuth2AuthenticationToken)
             .cast(OAuth2AuthenticationToken.class)
             .flatMap(authentication -> authorizedClient(exchange, authentication))
+            .onErrorResume(e -> Mono.empty())
```
Comment (mshima) [inline generators/server/templates/src/main/java/package/web/filter/OAuth2ReactiveRefreshTokensWebFilter.java.ejs:49]: It should redirect to the oauth2 login here.
If the token is expired, an error is returned.

**Comment by mshima** [comment]
Merging.
It fixes a problem with expired tokens, where the session get stuck in a broken state.
The refresh token still works correctly.

Reverts the old behavior of expired tokens for reactive and redirect to oauth2 login for non-reactive.

discussion can continue.

File: generators/server/templates/src/main/java/package/web/filter/OAuth2RefreshTokensWebFilter.java.ejs [hunk 3]
```diff
@@ -40,16 +48,39 @@ import org.springframework.web.filter.OncePerRequestFilter;
...
+                OAuth2AuthorizationRequestRedirectFilter.DEFAULT_AUTHORIZATION_REQUEST_BASE_URI
+            );
     }
 
     @Override
-    public void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain) throws IOException, ServletException {
+    public void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
+        throws IOException, ServletException {
         Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
         if ((authentication instanceof OAuth2AuthenticationToken)) {
```
Comment (Tcharl) [inline generators/server/templates/src/main/java/package/web/filter/OAuth2RefreshTokensWebFilter.java.ejs:73]: can remove the doubled parenthesis

**Pull request merged**